### PR TITLE
Fixes robotic cell death

### LINKED ISSUE #40802: Autophagocytosis Necrosis (flesh eating virus) damages prosthetic/robotic limbs.
what is says on the tin.

### PR DESCRIPTION
**Pull request by QualityVan**
:cl: QualityVan
fix: Diseases that damage bodyparts only work on organics now
/:cl:

Fixes #40802

### CODE REVIEW

File: code/modules/mob/living/damage_procs.dm [hunk 1]
```diff
@@ -264,7 +264,7 @@
 		update_stamina()
 
 // damage MANY bodyparts, in random order
-/mob/living/proc/take_overall_damage(brute = 0, burn = 0, stamina = 0, updating_health = TRUE)
+/mob/living/proc/take_overall_damage(brute = 0, burn = 0, stamina = 0, updating_health = TRUE, only_robotic = FALSE, only_organic = FALSE)
```
Comment (ShizCalev) [inline code/modules/mob/living/damage_procs.dm]: should probably just make this one arg for future proofing in-case we ever add more, `effected_bodypart_tissue` or something along those lines

Comment (ShizCalev) [inline code/modules/mob/living/damage_procs.dm]: should probably just make this one var that passes a status type, that way if we ever add more we won't have to add more arguments.

**Review by SpaceManiac — DISMISSED** [review]
What Shiz said, make it `required_status` or something and cancel if `required_status && status != required_status`.

**Comment by QualityVan** [comment]
Not feeling up to doing an entire iteration over all healing chems to make that work.

**Comment by SpaceManiac** [comment]
You won't need to change any lines you haven't already

**Comment by SpaceManiac** [comment]
I meant `required_status` as an argument to `take_overall_damage` instead of two arguments `only_robotic` and `only_organic` in case it wasn't clear.

**Comment by QualityVan** [comment]
That's incorrect, unless you're okay with leaving the take damage and heal damage procs nonparallel. Still, you're more than welcome to make the changes yourself, maintainers have access.

**Comment by SpaceManiac** [comment]
You appear to have not checked that box
```
 ! [remote rejected]       HEAD -> limbbiodamageonly (permission denied)
error: failed to push some refs to 'git@github.com:QualityVan/tgstation'
```

**Comment by QualityVan** [comment]
I definitely hit the box. Might be because the PR is closed?

**Pull request merged**